File: 250424/사는 지역/where-live.cpp
#include <iostream>
#include <string>
using namespace std;

class Person {
    public:
        string name;
        string address;
        string place;
        Person(string name, string address, string place) {
            this->name = name;
            this->address = address;
            this->place = place;
        }
        Person() {};
};

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n;
    string name, address, place;
    Person person[10];
    cin >> n;
    
    for(int i = 0; i < n; i++) {
        cin >> name >> address >> place;
        person[i] = Person(name, address, place);
    }

    int index = 0;
    for(int i = 0; i < n; i++) {
        if(person[i].name > person[index].name) {
            index = i;
        }
    }
    cout << "name " << person[index].name << "\n";
    cout << "addr " << person[index].address << "\n";
    cout << "city " << person[index].place << "\n";
    return 0;
}

File: 250424/사는 지역/where-live.java
import java.util.Scanner;
import java.util.Arrays;

class LivePlace {
    String name;
    String address;
    String place;

    public LivePlace(String name, String address, String place) {
        this.name = name;
        this.address = address;
        this.place = place;
    }
}
public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        LivePlace[] livePlaces = new LivePlace[10];
        int n = sc.nextInt();
        for(int i = 0; i < n; i++) {
            String name = sc.next();
            String address = sc.next();
            String place = sc.next();
            livePlaces[i] = new LivePlace(name, address, place);
        }
        int lastindex = 0;
        for(int i = 1; i < n; i++) {
            if(livePlaces[i].name.compareTo(livePlaces[lastindex].name) > 0) {
                lastindex = i;
            }
        }
        System.out.println("name " + livePlaces[lastindex].name);
        System.out.println("addr " + livePlaces[lastindex].address);
        System.out.println("city " + livePlaces[lastindex].place);
        sc.close();
    }
}